### PR TITLE
Reverse proxy with nginx

### PR DESCRIPTION
**Pull request by maxhelias**
Replace the reverse proxy with nginx

### CODE REVIEW

File: Dockerfile.h2-proxy [hunk 1]
```diff
@@ -10,8 +10,8 @@ RUN openssl req -new -passout pass:NotSecure -key server.key -out server.csr \
...
 
-FROM httpd:2.4-alpine
+FROM nginx:1.15-alpine
 
-COPY --from=0 server.key /usr/local/apache2/conf/server.key
-COPY --from=0 server.crt /usr/local/apache2/conf/server.crt
-COPY ./docker/httpd/httpd.conf /usr/local/apache2/conf/httpd.conf
+RUN mkdir -p /etc/nginx/ssl/
+COPY --from=0 server.key server.crt /etc/nginx/ssl/
+COPY ./docker/h2-proxy/default.conf /etc/nginx/conf.d/default.conf
```
Comment (dunglas) [inline Dockerfile.h2-proxy]: Missing blank line at the end

File: docker/h2-proxy/default.conf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+server {
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+
+    ssl_certificate /etc/nginx/ssl/server.crt;
+    ssl_certificate_key /etc/nginx/ssl/server.key;
+    ssl_session_cache  builtin:1000  shared:SSL:10m;
```
Comment (dunglas) [inline docker/h2-proxy/default.conf]: I wouldn't change the default value (as this container is only intended for dev, we can keep the config minimal).

File: Dockerfile.h2-proxy (hunk 1 above)
Comment (ajardin) [inline Dockerfile.h2-proxy]: Don't forget the newline at the end of file. 😉

**Comment by dunglas** [comment]
@teohhanhui, would you mean reviewing this one?

File: docker-compose.yaml [hunk 3]
```diff
@@ -35,6 +35,6 @@ services:
       context: .
       dockerfile: ./Dockerfile.h2-proxy
     volumes:
-      - ./docker/httpd/httpd.conf:/usr/local/apache2/conf/httpd.conf:ro
+      - ./docker/h2-proxy/default.conf:/etc/nginx/conf.d/default.conf:ro
```
Comment (ajardin) [inline docker-compose.yaml]: As the file seems to be copied during the image building, is it really needed?

Comment (dunglas) [reply docker-compose.yaml]: Yes. Copying is for prod, the volume is for dev (you change the file, reboot the service and it's taken into account without having to rebuild the image)

Comment (ajardin) [reply docker-compose.yaml]: That's my first guess. 😄 
Maybe we can add the same notice as for the nginx service (`# Comment out the next line in production`).

**Comment by maxhelias** [comment]
@dunglas it's good 😃

**Comment by B-Galati** [comment]
What about doing the same thing in the Dockerfile.nginx ?
Would it be wrong or impossible ?

**Comment by maxhelias** [comment]
@B-Galati It is not secure for use in production because it is a self-generated certificate.
We use the h2-proxy service only for development

**Comment by B-Galati** [comment]
@maxhelias Thanks for your answer.

Considering a development environment only, does it make sense to keep both `nginx` and `h2-proxy` service while `nginx` could do both ?

**Comment by maxhelias** [comment]
@B-Galati Yes, if you use it only in development you can configure your h2 directly in the nginx service.

**Comment by maxhelias** [comment]
With the current configuration, you can use it in both cases.
Just pay attention to the docker-compose's comments about the production before pushing the images

**Comment by dunglas** [comment]
Thanks @maxhelias!

**Pull request merged**